public/app/core/history/richHistoryStorageProvider.ts

```diff
@@ -0,0 +1,13 @@
+import RichHistoryLocalStorage from './richHistoryLocalStorage';
+import RichHistoryRemoteStorage from './richHistoryRemoteStorage';
+import RichHistoryStorage from './richHistoryStorage';
+
+const richHistoryLocalStorage = new RichHistoryLocalStorage();
+
+// To be implemented and switched with a feature flag.
+// @ts-ignore
+const richHistoryRemoteStorage = new RichHistoryRemoteStorage();
+
+export const getRichHistoryStorage = (): RichHistoryStorage => {
+  return richHistoryLocalStorage;
+};
```

Comment:
why not just this?
```suggestion
export const richHistoryStorage = new RichHistoryLocalStorage();
```

Reply:
Because it will be guarded by a feature flag. I would need to check at what point we get access to feature flags - is it accessible when the file is parsed?

Reply:
makes sense! not sure about when feature flags are available, but we can double check when we implement it 👍

Reply:
I think we will add it as soon as some of remote storage functionalities are implemented in the front end for testing.